tikv-client/src/main/java/com/pingcap/tikv/util/ChannelFactory.java

```diff
@@ -38,13 +67,24 @@ private ManagedChannel addrToChannel(String addressStr) {
     } catch (Exception e) {
       throw new IllegalArgumentException("failed to form address " + addressStr);
     }
-    // Channel should be lazy without actual connection until first call
-    // So a coarse grain lock is ok here
-    return ManagedChannelBuilder.forAddress(address.getHost(), address.getPort())
-        .maxInboundMessageSize(maxFrameSize)
-        .usePlaintext()
-        .idleTimeout(60, TimeUnit.SECONDS)
-        .build();
+
+    NettyChannelBuilder builder =
+        NettyChannelBuilder.forAddress(address.getHost(), address.getPort())
+            .maxInboundMessageSize(maxFrameSize)
+            .keepAliveWithoutCalls(true)
+            .idleTimeout(60, TimeUnit.SECONDS);
+
+    if (sslContextBuilder == null) {
+      return builder.usePlaintext().build();
+    } else {
+      SslContext sslContext = null;
+      try {
+        sslContext = sslContextBuilder.build();
+      } catch (SSLException e) {
+        return null;
+      }
+      return builder.sslContext(sslContext).build();
+    }
```

Comment:
I have a question: do we need to change address to `https` in SSL

Reply:
HTTPS or HTTP is the same here. `ChannelFactory` just needs the IP address to build a channel with the server. So whatever HTTPS or HTTP, it will work well.